src/Analyzer.JsonRuleEngine/JsonRuleEngine.cs

```diff
@@ -49,6 +49,51 @@ public static JsonRuleEngine Create(string rawRuleDefinitions, BuildILineNumberR
             return new JsonRuleEngine(ParseRuleDefinitions(rawRuleDefinitions), jsonLineNumberResolverBuilder);
         }
 
+        /// <summary>
+        /// Modifies the rules to run based on values defined in the configurations file.
+        /// </summary>
+        /// <param name="configuration">The template configurations to filter and analyze.</param>
+        public void FilterRules(string configuration)
```

Comment:
This function needs tests and should be moved outside of the context of the rule engines. It should be somewhere in the TemplateAnalyzer code since the configurations should cover all rules across the engines.

Reply:
Oh, I forgot about TTK, if we want this configuration file to work with those rules we also need to go through those rules

Reply:
Here's the info on how to suppress/select TTK rules: https://github.com/Azure/arm-ttk/tree/master/arm-ttk The examples show one rule at a time but I think multiple are possible

Reply:
Vera for TTK, because it is a cli argument, should this still be in here? / where does the loading of ttk rules happen in BPA?
Yanelis, Johnathon, I am confused by the decision. Are we agreeing that each rule engine will have their own FilterRules or it will be universal? This may get clearer based on where I move calling this function too.

Reply:
Others can chime in, but here's my opinion:
- An engine shouldn't run rules that aren't going to be reported, so the engine should filter the rules before analysis.
  - For this reason, I think filtering should stay in the context of the engine, since Analyzer.Core won't know how to filter.
- The way TTK runs currently is temporary, so I think we shouldn't put effort into filtering those results.
  - Instead, we should put our effort into creating the future PowerShell engine that TTK rules can be rewritten in.

Reply:
@yane3628 In the last discussion, you wanted me to move filtering into Analyzer.Core Template Analyzer from JsonRuleEngine without the addition of the interface. However, doing that means having to re-create/reference the RuleDefinition Schema in the JsonRuleEngine as when filtering, I look at the properties inside the rule schema. Leaving it inside the JsonRuleEngine and each engine having their own filtering makes the most sense in my opinion (like Johnathon mentioned above).